### PR TITLE
healthcheck: wait for systemd operations

### PR DESCRIPTION
**Pull request by vrothberg**
Make sure to wait for the systemd operations to finish when
starting/stopping healtcheck timers and services.  Also make
sure to stop the timer before the service to avoid a race
with the timer.

[NO NEW TESTS NEEDED] since it is a non-functional change and existing
tests are expected to pass.

Signed-off-by: Valentin Rothberg <vrothberg@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
Fix a potential race when cleaning up systemd health-check units and timers.
```

### CODE REVIEW

**Comment by vrothberg** [comment]
@msekletar @mheon @rhatdan PTAL

**Comment by rhatdan** [comment]
LGTM

File: libpod/healthcheck_linux.go [hunk 1]
```diff
@@ -46,6 +47,17 @@ func (c *Container) createTimer() error {
 	return nil
 }
 
+// Wait for a message on the channel.  Throw an error if the message is not "done".
+func systemdOpSuccessful(c chan string) error {
+	msg := <-c
+	switch msg {
+	case "done":
```
Comment (TomSweeneyRedHat) [inline libpod/healthcheck_linux.go:54]: is it guaranteed to be lower case "done"?

**Comment by TomSweeneyRedHat** [comment]
one question, otherwise LGTM.  Would like a head nod from one of the requested folks, especially @msekletar

File: libpod/healthcheck_linux.go (hunk 1 above)
Comment (vrothberg) [reply libpod/healthcheck_linux.go:54]: Yes, that's guaranteed.

**Review by flouthoc — APPROVED** [review]
/lgtm
/approve

**Pull request merged**